sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java

```diff
@@ -173,6 +174,34 @@ Mono<Response<Void>> recognizeWithResponseInternal(CallMediaRecognizeOptions rec
 
                 return contentsInternal.recognizeWithResponseAsync(callConnectionId, recognizeRequest, context);
 
+            } else if (recognizeOptions instanceof CallMediaRecognizeChoiceOptions) {
+                CallMediaRecognizeChoiceOptions choiceRecognizeOptions = (CallMediaRecognizeChoiceOptions) recognizeOptions;
+
+                RecognizeOptionsInternal recognizeOptionsInternal = new RecognizeOptionsInternal()
+                    .setChoices(choiceRecognizeOptions.getRecognizeChoices())
+                    .setInterruptPrompt(recognizeOptions.isInterruptPrompt())
+                    .setTargetParticipant(CommunicationIdentifierConverter.convert(recognizeOptions.getTargetParticipant()));
+
+                recognizeOptionsInternal.setInitialSilenceTimeoutInSeconds((int) recognizeOptions.getInitialSilenceTimeout().getSeconds());
+
+                PlaySourceInternal playSourceInternal = null;
+                if (recognizeOptions.getPlayPrompt() != null) {
+                    PlaySource playSource = recognizeOptions.getPlayPrompt();
+                    if (playSource instanceof FileSource) {
+                        playSourceInternal = getPlaySourceInternalFromFileSource((FileSource) playSource);
+                    } else if (playSource instanceof TextSource) {
```

Comment:
Are we supporting this in the Play Operation? should we add it?

Reply:
Is this supported on the SDK's Play Operation? should we add it?

Reply:
Yes, the SDK play operation support the play text source.